src/api/app/datatables/project_datatable.rb

```diff
@@ -30,10 +30,12 @@ def get_raw_records
   # rubocop:enable Naming/AccessorMethodName
 
   def data
-    records.map do |record|
+    attrib_type_category = AttribType.joins(:attrib_namespace).where(attrib_namespace: { name: 'OBS' }).find_by(attrib_types: { name: 'QualityCategory' })[:id]
+    attribs_join_condition = "LEFT OUTER JOIN attribs ON (attribs.project_id = projects.id AND attribs.attrib_type_id = #{attrib_type_category}) " \
```

Comment:
You can do left outer joins with active record too right?

Reply:
> You can do left outer joins with active record too right?

Sure. The `attribs.project_id = projects.id AND attribs.attrib_type_id` part of the condition can be done with an `includes` or a `left_outer_joins`.

But I didn't find any other way to modify that join condition to add the `AND attribs.attrib_type_id = #{attrib_type_category}` part.

Reply:
That returns the attributes. Good.

We still need the attrirbute value. From there, any way to return the attribute value of that quality attribute?

Reply:
It worked for me. Nice! Thanks!

In my development environment it works fine. :+1:

With `records.left_outer_joins(quality_attribs: :values)`, the resulting SQL is:

```
SELECT projects.*, attrib_values.value AS attrib_value
FROM `projects` LEFT OUTER JOIN `attribs` ON
  `attribs`.`attrib_type_id` IN (SELECT `attrib_types`.`id` FROM `attrib_types` INNER JOIN `attrib_namespaces` `attrib_namespace` ON `attrib_namespace`.`id` = `attrib_types`.`attrib_namespace_id` WHERE `attrib_namespace`.`name` = 'OBS' AND `attrib_types`.`name` = 'QualityCategory')
    AND `attribs`.`project_id` = `projects`.`id`
LEFT OUTER JOIN `attrib_values` ON `attrib_values`.`attrib_id` = `attribs`.`id`
WHERE `projects`.`id` != 0 AND NOT (name rlike '^home:.+')
ORDER BY projects.name ASC
LIMIT 25
OFFSET 0
```

I guess that when you mentioned `left_inner_joins` you meant `left_outer_joins`.